packages/extension/src/components/pages/AddNewTrustline/AddNewTrustline.tsx

```diff
@@ -231,24 +245,76 @@ export const AddNewTrustline: FC = () => {
       })
         .then((transactionHash) => {
           setTransaction(TransactionStatus.Success);
-          chrome.runtime.sendMessage<
-            ReceiveSetTrustlineBackgroundMessage | ReceiveSetTrustlineBackgroundMessageDeprecated
-          >(createMessage({ transactionHash }));
+          if (!params.inAppCall) {
+            chrome.runtime.sendMessage<
+              ReceiveSetTrustlineBackgroundMessage | ReceiveSetTrustlineBackgroundMessageDeprecated
+            >(createMessage({ transactionHash }));
+          }
         })
         .catch((e) => {
           setErrorRequestRejection(e.message);
           setTransaction(TransactionStatus.Rejected);
-          chrome.runtime.sendMessage<
-            ReceiveSetTrustlineBackgroundMessage | ReceiveSetTrustlineBackgroundMessageDeprecated
-          >(createMessage({ transactionHash: undefined, error: e }));
+          if (!params.inAppCall) {
+            chrome.runtime.sendMessage<
+              ReceiveSetTrustlineBackgroundMessage | ReceiveSetTrustlineBackgroundMessageDeprecated
+            >(
+              createMessage({
+                transactionHash: undefined,
+                error: e
+              })
+            );
+          }
         });
     }
-  }, [setTrustline, createMessage, params.limitAmount, params.fee, params.flags, params.memos]);
+  }, [
+    setTrustline,
+    createMessage,
+    params.limitAmount,
+    params.fee,
+    params.inAppCall,
+    params.flags,
+    params.memos
+  ]);
+
+  const handleTrustlineSubmit = (
+    issuer: string,
+    token: string,
+    limit: string,
+    showForm: boolean,
+    isParamsMissing: boolean
+  ) => {
+    setParams({
+      limitAmount: {
+        currency: token,
+        issuer: issuer,
+        value: limit
+      },
+      fee: params.fee,
+      id: params.id,
+      memos: params.memos,
+      flags: params.flags,
+      showForm: showForm,
+      inAppCall: true
+    });
+
+    setIsParamsMissing(isParamsMissing);
+  };
+
+  if (params.showForm) {
+    return <StepForm onTrustlineSubmit={handleTrustlineSubmit} />;
+  }
 
   if (isParamsMissing) {
```

Comment:
This can be simplified like this:
```suggestion
  if (isParamsMissing && !params.inAppCall) {
```

Reply:
We still want to render what's below even if it's an in app call